enclave-boundary/src/untrusted.rs

```diff
@@ -46,16 +46,42 @@ pub fn make_variable_length_ecall(
                 &mut outbuf_used,
                 &mut outbuf_retry_id,
             )
-        } {
+        };
+        match result {
             sgx_status_t::SGX_SUCCESS => match retval {
                 sgx_status_t::SGX_ERROR_OUT_OF_MEMORY => continue,
                 sgx_status_t::SGX_SUCCESS => {
                     outbuf.truncate(outbuf_used);
                     break Ok(outbuf);
                 }
+                other_retval if is_fatal_sgx_status(other_retval) => {
+                    panic!(
+                        "Enclave reported fatal error: ecall retval: {:?}. Panicking to restart.",
+                        retval
+                    );
+                }
                 other_retval => break (Err(other_retval)),
             },
+            status if is_fatal_sgx_status(status) => {
+                panic!(
+                    "Enclave reported fatal error: ecall returned {:?}. Panicking to restart.",
```

Comment:
Nit but the restart behavior is a side-effect of how we run this in our k8 setup, it is not behavior that is guaranteed.

```suggestion
                    "Enclave reported fatal error: ecall returned {:?}",
```